glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -555,17 +542,16 @@ open class GleanInternalAPI internal constructor () {
 
         if (!getUploadEnabled()) {
             Log.e(LOG_TAG, "Glean must be enabled before submitting pings.")
-            return
+        } else {
+            LibGleanFFI.INSTANCE.glean_submit_ping_by_name(
+                    pingName,
+                    reason
+            )
         }
 
-        val submittedPing = LibGleanFFI.INSTANCE.glean_submit_ping_by_name(
-            pingName,
-            reason
-        ).toBoolean()
-
-        if (submittedPing) {
-            PingUploadWorker.enqueueWorker(applicationContext)
-        }
+        // Enqueue the PingUploadWorker, this does not depend on wether upload is enabled,
+        // because there might be deletion-request ping in that case.
```

Comment:
Do we really need to change this? We would still kick off the uploader when glean starts, right?

Reply:
We need ping uploader to be kicked even if upload is disabled. If we don't change this, that doesn't happen.